idact/detail/auth/authenticate.py

```diff
@@ -111,7 +111,7 @@ def authenticate(host: str,
                                     after authentication.
 
     """
-    if config.auth not in [AuthMethod.ASK, AuthMethod.PUBLIC_KEY]:
+    if config.auth not in AuthMethod.__members__.values():
```

Comment:
I think that only one option that this condition can be met is when config.auth is None. In this case error saying that this is not implemented is quite not informative.

Reply:
https://en.wikipedia.org/wiki/Assertion_(software_development)